### PR TITLE
Update kernel to v5.15.48.1

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Update kernel to v5.15.48.1

###### Change Log  
- Update kernel to v5.15.48.1

###### Does this affect the toolchain?  
**YES**

###### Test Methodology
- Pipeline : 
- ARM: [#204616](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=204616&view=results) Kernel built, pipeline failed due to kpatch not building (unrelated)
- AMD: [#204182](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=204182&view=results) built successfully

### CODE REVIEW

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 1]
```diff
@@ -67,14 +67,14 @@ set -e
 #
 cd /sources
 
-echo Linux-5.15.45.1 API Headers
-tar xf kernel-5.15.45.1.tar.gz
-pushd CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.45.1
+echo Linux-5.15.48.1 API Headers
```
Comment (mfrw) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:70]: In my opinion we should reduce typing as much as possible :)
How about:
```suggestion
KERNEL_VERSION="5.15.48.1"
echo Linux-${KERNEL_VERSION} API Headers
```

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 2]
```diff
@@ -67,14 +67,14 @@ set -e
...
+tar xf kernel-5.15.48.1.tar.gz
+pushd CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.48.1
```
Comment (mfrw) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:72]:
```suggestion
tar xf kernel-${KERNEL_VERSION}.tar.gz
pushd CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-${KERNEL_VERSION}
```

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 3]
```diff
@@ -67,14 +67,14 @@ set -e
...
 make mrproper
 make headers
 cp -rv usr/include/* /usr/include
 popd
-rm -rf CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.45.1
+rm -rf CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.48.1
```
Comment (mfrw) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:77]:
```suggestion
rm -rf CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-${KERNEL_VERSION}
```

File: toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh [hunk 4]
```diff
@@ -114,14 +114,14 @@ rm -rf gcc-11.2.0
 
 touch $LFS/logs/temptoolchain/status_gcc_pass1_complete
 
-echo Linux-5.15.45.1 API Headers
-tar xf kernel-5.15.45.1.tar.gz
-pushd CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.45.1
+echo Linux-5.15.48.1 API Headers
```
Comment (mfrw) [inline toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh:117]:
```suggestion
KERNEL_VERSION="5.15.48.1"
echo Linux-${KERNEL_VERSION} API Headers
```

File: toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh [hunk 5]
```diff
@@ -114,14 +114,14 @@ rm -rf gcc-11.2.0
...
+tar xf kernel-5.15.48.1.tar.gz
+pushd CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.48.1
```
Comment (mfrw) [inline toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh:119]:
```suggestion
tar xf kernel-${KERNEL_VERSION}.tar.gz
pushd CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-${KERNEL_VERSION}
```

File: toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh [hunk 6]
```diff
@@ -114,14 +114,14 @@ rm -rf gcc-11.2.0
...
 make mrproper
 make headers
 cp -rv usr/include/* /tools/include
 popd
-rm -rf CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.45.1
+rm -rf CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-5.15.48.1
```
Comment (mfrw) [inline toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh:124]:
```suggestion
rm -rf CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-${KERNEL_VERSION}
```

File: toolkit/scripts/update_kernel.sh [hunk 7]
```diff
@@ -111,14 +111,8 @@ function update_toolchain_scripts {
...
         file=$TOOLCHAIN_FOLDER/container/$script
-        PATTERN="echo Linux-.* API Headers"
-        REPLACE="echo Linux-$VERSION API Headers"
-        sed -i "s/$PATTERN/$REPLACE/" $file
-        PATTERN="tar xf $FILE_SIGNATURE_PATTERN.*.tar.gz"
-        REPLACE="tar xf $TARBALL_NAME"
-        sed -i "s/$PATTERN/$REPLACE/" $file
-        PATTERN="CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-.*"
-        REPLACE="CBL-Mariner-Linux-Kernel-rolling-lts-mariner-2-$VERSION"
+        PATTERN="KERNEL_VERSION=\"$OLD_VERSION\""
```
[the remark this answers is not in the document]
Comment (neha170) [reply toolkit/scripts/update_kernel.sh:114]: Yep, great suggestion by Falak

[the remark this answers is not in the document]
Comment (mfrw) [reply toolkit/scripts/update_kernel.sh:114]: FWIW,
I had completely missed updating this file. so, Yep great catch indeed :)

**Pull request merged**